### PR TITLE
add tidb-3.0 compatibility document

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 
add tidb-3.0 compatibility document

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
 ## Statistics information
 If you want to know how TiSpark could benefit from TiDB's statistic information, read more [here](./docs/userguide.md).
 
+## Compatibility with tidb-3.0
+### View
+TiDB start to support `view` from `tidb-3.0`.
```
Comment (zhexuany) [inline README.md]: s/start/starts

File: README.md [hunk 2]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
...
+## Compatibility with tidb-3.0
+### View
+TiDB start to support `view` from `tidb-3.0`. 
+
+TiSpark currently does not support `view` and will filter all views from catalog. Users cannot select data from views through TiSpark.
+
+### Table Partition
+`tidb-3.0` supports both `Range Partition` and `Hash Partition`. 
+
+TiSpark currently does support `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark.
```
Comment (birdstorm) [inline README.md]:
```suggestion
TiSpark currently **support** `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark. 
```

File: README.md [hunk 3]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
 ## Statistics information
 If you want to know how TiSpark could benefit from TiDB's statistic information, read more [here](./docs/userguide.md).
 
+## Compatibility with tidb-3.0
+### View
+TiDB start to support `view` from `tidb-3.0`. 
+
+TiSpark currently does not support `view` and will filter all views from catalog. Users cannot select data from views through TiSpark.
```
Comment (birdstorm) [inline README.md]:
```suggestion
TiSpark currently **does not support** `view`. Users will not be able to observe or access data through views with TiSpark.
```

File: README.md [hunk 4]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
...
+
+### Table Partition
+`tidb-3.0` supports both `Range Partition` and `Hash Partition`. 
+
+TiSpark currently does support `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark. 
+
+In most cases TiSpark will use full table scan. Only in such cases TiSpark will apply partition pruning:
```
Comment (zhexuany) [inline README.md]: s/Only in such cases TiSpark will apply partition pruning/Only in the following cases TiSpark will apply partition pruning to build a better plan

File: README.md [hunk 5]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
...
+- table is `Range Partition`
+- partition expression only contains function `year`.
```
Comment (zhexuany) [inline README.md]: partition expression is column or year expression whose child is column.

**Comment by marsishandsome** [comment]
@birdstorm @zhexuany PTAL again

File: README.md [hunk 6]
```diff
@@ -166,6 +166,21 @@ Time Zone can be set by using `-Duser.timezone` system property, e.g. `-Duser.ti
...
+TiSpark currently **does not support** `view`. Users will not be able to observe or access data through views with TiSpark.
+
+### Table Partition
+`tidb-3.0` supports both `Range Partition` and `Hash Partition`. 
+
+TiSpark currently **supports** `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark. 
+
+In most cases TiSpark will use full table scan. Only in the following cases TiSpark will apply partition pruning:
+- table is `Range Partition`
+- partition expression is a `column` expression or `year` expression whose child is a column expression
```
Comment (zhexuany) [inline README.md]: and its type is datetime or string literal but can be parsed as datetime.

**Pull request merged**